common/network-common/src/main/java/org/apache/spark/network/util/NettyUtils.java

```diff
@@ -95,6 +111,38 @@ public static String getRemoteAddress(Channel channel) {
     return "<unknown remote>";
   }
 
+  /**
+   * Returns the default number of threads for both the Netty client and server thread pools.
+   * If numUsableCores is 0, we will use Runtime get an approximate number of available cores.
+   */
+  public static int defaultNumThreads(int numUsableCores) {
+    final int availableCores;
+    if (numUsableCores > 0) {
+      availableCores = numUsableCores;
+    } else {
+      availableCores = Runtime.getRuntime().availableProcessors();
+    }
+    return Math.min(availableCores, MAX_DEFAULT_NETTY_THREADS);
+  }
+
+  /**
+   * Returns the lazily created shared pooled ByteBuf allocator for the specified allowCache
+   * parameter value.
+   */
+  public static synchronized PooledByteBufAllocator getSharedPooledByteBufAllocator(
+      boolean allowDirectBufs,
+      boolean allowCache) {
+    final int index = allowCache ? 0 : 1;
```

Comment:
I am not sure if it is a good idea to piggyback on allowCache to determine whether is a client/server pooled allocator. Maybe use another variable?

Reply:
Thanks for reviewing it!

Introducing a new parameter like `isClient` would just hide the that the value of the  allowCache is differs but at the callers this level of abstraction is already introduced. I mean with calling `createPooledByteBufAllocator`, like: 
[TransportServer.java#L77](https://github.com/apache/spark/blob/ae8fa8e81510c69f84444c368cad25788f5423b4/common/network-common/src/main/java/org/apache/spark/network/server/TransportServer.java#L77 )

So I would keep this (or change createPooledByteBufAllocator too).

Reply:
I don't think that you need to change createPooledByteBufAllocator if you change this method. allowCache is used to determine whether to create a pool with or without cache in createPooledByteBufAllocator. This is different from getSharedPooledByteBufAllocator which is using allowCache to determine this plus, it is also using allowCache to determine whether to return a pooledAllocator for a client or a server.

You may still want to keep this change if allowCache is the only thing that distinguishes the two pooled allocators but if it is not then it may be better to create another variable.